src/redux/validation/validation.listeners.tsx

```diff
@@ -0,0 +1,133 @@
+import {Modal} from 'antd';
+
+import {ExclamationCircleOutlined} from '@ant-design/icons';
+
+import {isAnyOf} from '@reduxjs/toolkit';
+
+import {isEmpty, isEqual} from 'lodash';
+
+import {AppListenerFn} from '@redux/listeners/base';
+import {updateK8sVersion, updateProjectK8sVersion} from '@redux/reducers/appConfig';
+import {currentConfigSelector} from '@redux/selectors';
+import {clusterResourceMapSelector} from '@redux/selectors/resourceMapSelectors';
+import {loadClusterResources, reloadClusterResources} from '@redux/thunks/cluster';
+import {downloadK8sSchema} from '@redux/thunks/downloadK8sSchema';
+import {setRootFolder} from '@redux/thunks/setRootFolder';
+
+import {doesSchemaExist} from '@utils/index';
+
+import {activeProjectSelector, kubeConfigContextSelector} from '@shared/utils/selectors';
+
+import {setConfigK8sSchemaVersion, toggleOPARules, toggleValidation} from './validation.slice';
+import {loadValidation, validateResources} from './validation.thunks';
+
+const loadListener: AppListenerFn = listen => {
+  listen({
+    matcher: isAnyOf(
+      loadClusterResources.fulfilled,
+      reloadClusterResources.fulfilled,
+      setRootFolder.fulfilled,
```

Comment:
do we have to load the validator for `loadClusterResources` and `reloadClusterResources` ?
maybe for the `loadClusterResources` it makes sense if it's triggered by the quick cluster loading?